src/Umbraco.Core/PropertyEditors/ValueConverters/SliderValueConverter.cs

```diff
@@ -11,77 +8,73 @@ namespace Umbraco.Core.PropertyEditors.ValueConverters
+            var sourceString = source?.ToString();
+            if (string.IsNullOrEmpty(sourceString))
+            {
                 return null;
+            }
 
-            if (IsRangeDataType(propertyType.DataType.Id))
+            if (IsRangeDataType(propertyType))
             {
-                var rangeRawValues = source.ToString().Split(',');
+                // Return range
+                var rangeRawValues = sourceString.Split(',');
                 var minimumAttempt = rangeRawValues[0].TryConvertTo<decimal>();
-                var maximumAttempt = rangeRawValues[1].TryConvertTo<decimal>();
 
-                if (minimumAttempt.Success && maximumAttempt.Success)
+                if (rangeRawValues.Length == 1 && minimumAttempt.Success)
+                {
+                    // Configuration is probably changed from single to range, return range with same min/max
+                    return new Range<decimal>
+                    {
+                        Minimum = minimumAttempt.Result,
+                        Maximum = minimumAttempt.Result
+                    };
+                }
+                else if (rangeRawValues.Length == 2)
                 {
-                    return new Range<decimal> { Maximum = maximumAttempt.Result, Minimum = minimumAttempt.Result };
+                    var maximumAttempt = rangeRawValues[1].TryConvertTo<decimal>();
+                    if (maximumAttempt.Success)
+                    {
+                        return new Range<decimal>
+                        {
+                            Minimum = minimumAttempt.Result,
+                            Maximum = maximumAttempt.Result
+                        };
+                    }
+                }
+            }
+            else
+            {
+                // Return single value
+                var valueAttempt = sourceString.TryConvertTo<decimal>();
+                if (valueAttempt.Success)
+                {
+                    return valueAttempt.Result;
                 }
             }
-
-            var valueAttempt = source.ToString().TryConvertTo<decimal>();
-            if (valueAttempt.Success)
-                return valueAttempt.Result;
 
             // Something failed in the conversion of the strings to decimals
             return null;
-
-        }
-
-        /// <summary>
-        /// Discovers if the slider is set to range mode.
-        /// </summary>
-        /// <param name="dataTypeId">
-        /// The data type id.
-        /// </param>
-        /// <returns>
-        /// The <see cref="bool"/>.
-        /// </returns>
-        private bool IsRangeDataType(int dataTypeId)
-        {
-            // GetPreValuesCollectionByDataTypeId is cached at repository level;
-            // still, the collection is deep-cloned so this is kinda expensive,
-            // better to cache here + trigger refresh in DataTypeCacheRefresher
-            // TODO: this is cheap now, remove the caching
```

Comment:
This isn't the case anymore in V8 (as the TODO comment confirms), so lets keep it simple and remove this unnecessary caching.